core/src/main/java/com/rudderstack/android/sdk/core/RudderDeviceInfo.java

```diff
@@ -69,4 +86,13 @@ String getAdvertisingId() {
     boolean isAdTrackingEnabled() {
         return this.adTrackingEnabled;
     }
+
+    void clearAdvertisingId() {
+        this.advertisingId = null;
+        this.adTrackingEnabled = (this.advertisingId != null);
+        if (RudderClient.getApplication() != null) {
+            RudderPreferenceManager preferenceManager = RudderPreferenceManager.getInstance(RudderClient.getApplication());
+            preferenceManager.clearAdvertisingId();
```

Comment:
As we get the `preferenceManager` object as an argument on the constructor, let's utilise that.

Reply:
agreed, just pushed the changes for the same.